lib/libzfs/os/linux/libzfs_mount_os.c

```diff
@@ -377,7 +377,9 @@ int
 do_unmount(const char *mntpt, int flags)
 {
 	if (!libzfs_envvar_is_set("ZFS_MOUNT_HELPER")) {
-		return (umount2(mntpt, flags));
+		int rv = umount2(mntpt, flags);
+
+		return (rv < 0 ? errno : rv);
```

Comment:
I think it would be slightly more clear to use:
`return (rv < 0 ? errno : 0);`

Reply:
fixed